app/models/user.php

```diff
@@ -184,4 +301,134 @@ public function memberSince()
         }
         return "{$registration_date} {$time_label}";
     }
+
+    public function editStatus()
+    {
+        try {
+            $db = DB::conn();
+            $db->begin();
+
+            $params = array(
+                'status' => BANNED, 
+                'id' => $this->user_id
+            );
+
+            $param = array(
+                'status' => ACTIVE, 
+                'id' => $this->user_id
+            );
+            
+            if ($this->current_status == ACTIVE) {
+                $update = $db->query('UPDATE user SET status= :status WHERE id=:id', $params);
+            } else {
+                $update = $db->query('UPDATE user SET status= :status WHERE id=:id', $param);
+            }
+
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
+
+    public function changePassword()
+    {
+        $this->validate();
+
+        if ($this->hasError()) {
+            throw new ValidationException('Error in Changing Password');
+        }
+
+        try {
+            $db = DB::conn();
+            $db->begin();
+
+            $params = array(
+                'password' => $this->mc_encrypt($this->password, ENCRYPTION_KEY),
+                'id'       => $_SESSION['userid']);
+
+            $update = $db->query('UPDATE user SET password = :password WHERE id = :id', $params);
+
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
+
+    //gets the top likers of the user
+    public function topLikers()
+    {
+        $users = array();
+        $db = DB::conn();
+
+        $rows = $db->rows('SELECT u.username AS Liker, COUNT(l.user_id) as Number_of_likes 
+            FROM like_monitor l 
+            INNER JOIN comment c on c.id = l.comment_id 
+            INNER JOIN user u on l.user_id = u.id 
+            WHERE c.user_id=? GROUP BY Liker ORDER BY Number_of_likes DESC LIMIT 5', 
+            array($_SESSION['userid']));
+        
+        foreach ($rows as $row) {
+             $users[] = new User($row);
+        }
+    
+        return $users;
+    }
+
+    //gets top commentors of the user
+    public function topCommentors()
+    {
+        $users = array();
+        $db = DB::conn();
+
+        $rows = $db->rows('SELECT a.username AS Commentor, COUNT(c.id) AS Number_of_comments FROM comment c 
+            INNER join thread t on t.id=c.thread_id 
+            INNER join user u on u.id=t.user_id 
+            INNER JOIN user a on a.id=c.user_id
+            WHERE u.id = ? GROUP BY c.user_id ORDER BY Number_of_comments DESC LIMIT 5',
+            array($_SESSION['userid']));
+
+        foreach ($rows as $row) {
+             $users[] = new User($row);
+        }
+
+        return $users;
+    }
+
+    // Encrypt Function
+    public function mc_encrypt($encrypt, $key)
```

Comment:
encrypt/decrypt method seems like a helper function, please remove it from this class as this is not related to the user :bow: